compiler/src/utils/filepath.re

```diff
@@ -85,6 +87,44 @@ module String = {
     String.capitalize_ascii(name);
   };
 
+  let normalize_separators = path =>
+    if (Sys.unix) {
+      path;
+    } else {
+      // If we aren't on a Unix-style system, convert `\\` separators to `/`
+      // This is needed because using `Filename` from OCaml stdlib doesn't get along with Fp
+      let windows_sep = Str.regexp("\\");
+      let normal_sep = "/";
+      Str.global_replace(windows_sep, normal_sep, path);
+    };
+
+  // TODO(#216): We should consider switching to type safe Fp.t where ever filepaths are used
+  let concat = (p1, p2) => {
+    normalize_separators(Filename.concat(p1, p2));
+  };
+
+  // TODO(#216): We should consider switching to type safe Fp.t where ever filepaths are used
+  let dirname = path => {
+    normalize_separators(Filename.dirname(path));
+  };
+
+  // TODO(#216): We should consider switching to type safe Fp.t where ever filepaths are used
+  let basename = Filename.basename;
+
+  // TODO: This is poorly named
```

Comment:
Do we want to just rename now?

Reply:
Yeah. Any suggestions? It's like `non_module_relative_path`? 🤣

Reply:
Could go the OCaml approach and name it `is_really_relative` 😛

Reply:
God no

Reply:
Let's skip for now. We probably want to change the way modules are specified when we completely change the module system.